packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -215,9 +215,16 @@ class OverlayState extends State<Overlay> {
   }
 
   void _remove(OverlayEntry entry) {
-    setState(() {
-      _entries.remove(entry);
+    assert(() {
+      if (!mounted)
+        assert(_entries.remove(entry));
```

Comment:
Yeah, but AFAICT (a) n is probably small (how many OverlayEntries are there likely to be?), and (b) we don't remove an OverlayEntry very often - maybe once in a frame.